test/claw_reference_test.cmake

```diff
@@ -476,41 +543,49 @@ function(claw_add_advanced_test)
     add_executable(${EXECUTABLE_TRANSFORMED_CPU} EXCLUDE_FROM_ALL
       "${OUTPUT_FILE_CPU}" "${OUTPUT_FILE_EXTRA_CPU}" "${OUTPUT_MAIN_CPU}")
     target_compile_definitions(${EXECUTABLE_TRANSFORMED_CPU} PRIVATE -D_CLAW)
-    add_executable(${EXECUTABLE_TRANSFORMED_GPU} EXCLUDE_FROM_ALL
-      "${OUTPUT_FILE_GPU}" "${OUTPUT_FILE_EXTRA_GPU}" "${OUTPUT_MAIN_GPU}")
-    target_compile_definitions(${EXECUTABLE_TRANSFORMED_GPU} PRIVATE -D_CLAW)
+    add_executable(${EXECUTABLE_TRANSFORMED_ACC} EXCLUDE_FROM_ALL
+      "${OUTPUT_FILE_ACC}" "${OUTPUT_FILE_EXTRA_ACC}" "${OUTPUT_MAIN_ACC}")
+    target_compile_definitions(${EXECUTABLE_TRANSFORMED_ACC} PRIVATE -D_CLAW)
+    add_executable(${EXECUTABLE_TRANSFORMED_OMP} EXCLUDE_FROM_ALL
+      "${OUTPUT_FILE_OMP}" "${OUTPUT_FILE_EXTRA_OMP}" "${OUTPUT_MAIN_OMP}")
+    target_compile_definitions(${EXECUTABLE_TRANSFORMED_OMP} PRIVATE -D_CLAW)
   else()
     # Build the original code and the transformed code
     add_executable(${EXECUTABLE_ORIGINAL} EXCLUDE_FROM_ALL
       "${ORIGINAL_FILE}" "${MAIN_F90}")
     add_executable(${EXECUTABLE_TRANSFORMED_CPU} EXCLUDE_FROM_ALL
       "${OUTPUT_FILE_CPU}" "${OUTPUT_MAIN_CPU}")
     target_compile_definitions(${EXECUTABLE_TRANSFORMED_CPU} PRIVATE -D_CLAW)
-    add_executable(${EXECUTABLE_TRANSFORMED_GPU} EXCLUDE_FROM_ALL
-      "${OUTPUT_FILE_GPU}" "${OUTPUT_MAIN_GPU}")
-    target_compile_definitions(${EXECUTABLE_TRANSFORMED_GPU} PRIVATE -D_CLAW)
+    add_executable(${EXECUTABLE_TRANSFORMED_ACC} EXCLUDE_FROM_ALL
+      "${OUTPUT_FILE_ACC}" "${OUTPUT_MAIN_ACC}")
+    target_compile_definitions(${EXECUTABLE_TRANSFORMED_ACC} PRIVATE -D_CLAW)
+    add_executable(${EXECUTABLE_TRANSFORMED_OMP} EXCLUDE_FROM_ALL
+      "${OUTPUT_FILE_OMP}" "${OUTPUT_MAIN_OMP}")
+    target_compile_definitions(${EXECUTABLE_TRANSFORMED_OMP} PRIVATE -D_CLAW)
   endif()
 
   # Target for the transformation
   add_custom_target(
     transform-${claw_add_advanced_test_NAME}
-    DEPENDS ${EXECUTABLE_TRANSFORMED_CPU} ${EXECUTABLE_TRANSFORMED_GPU}
+    DEPENDS ${EXECUTABLE_TRANSFORMED_CPU} ${EXECUTABLE_TRANSFORMED_ACC} ${EXECUTABLE_TRANSFORMED_OMP}
   )
 
-  add_dependencies(${EXECUTABLE_TRANSFORMED_GPU} ${EXECUTABLE_ORIGINAL})
+  add_dependencies(${EXECUTABLE_TRANSFORMED_ACC} ${EXECUTABLE_ORIGINAL})
+  add_dependencies(${EXECUTABLE_TRANSFORMED_OMP} ${EXECUTABLE_ORIGINAL})
   add_dependencies(${EXECUTABLE_TRANSFORMED_CPU} ${EXECUTABLE_ORIGINAL})
-  add_dependencies(${EXECUTABLE_TRANSFORMED_CPU} ${EXECUTABLE_TRANSFORMED_GPU})
+  add_dependencies(${EXECUTABLE_TRANSFORMED_CPU} ${EXECUTABLE_TRANSFORMED_ACC})
+  add_dependencies(${EXECUTABLE_TRANSFORMED_ACC} ${EXECUTABLE_TRANSFORMED_OMP})
```

Comment:
This is what I meant! So now the three targets in a single test are done sequentially but all the rest can still be executed in parallel!